src/tests/ftest/util/command_utils.py

```diff
@@ -352,6 +362,55 @@ def stop(self):
             self._process = None
 
 
+class DaosCommand(ExecutableCommand):
+    """A class for similar daos command line tools."""
+
+    def __init__(self, namespace, command, path=""):
+        """Create DaosCommand object.
+        Specific type of command object built so command str returns:
+            <command> <options> <request> <action/subcommand> <options>
+        Args:
+            namespace (str): yaml namespace (path to parameters)
+            command (str): string of the command to be executed.
+            path (str): path to location of daos command binary.
+        """
+        super(DaosCommand, self).__init__(namespace, command, path)
+        self.request = BasicParameter(None)
+        self.action = BasicParameter(None)
+        self.action_command = None
+
+    def _get_action_command(self):
+        """Assign a command object for the specified request and action."""
+        self.action_command = None
+
+    def get_param_names(self):
+        """Get a sorted list of DaosCommand parameter names."""
+        names = self.get_attribute_names(FormattedParameter)
+        names.extend(["request", "action"])
+        return names
+
+    def get_params(self, test):
+        """Get values for all of the command params from the yaml file.
+
+        Args:
+            test (Test): avocado Test object
+        """
+        super(DaosCommand, self).get_param_names(test)
+        self._get_action_command()
+
+    def get_str_param_names(self):
+        """Get a sorted list of the names of the command attributes.
+
+        Returns:
+            list: a list of class attribute names used to define parameters
+                for the command.
+
+        """
+        names = self.get_param_names()
+        names[-1] = "action_command"
```

Comment:
Since the `DmgCommand._get_action_command()` method can yield `self.action_command = None` this code should handle this case by always removing "action" for the name list and **only** adding "action_command" if it is defined.

Reply:
Oh good catch. But, if we always remove action, then it wouldn't add an action to the command in case action_command is not defined. We should only remove action when action_command is defined.

Reply:
Correct, I recant my comment about always removing "action"

Reply:
I'll change this to this from your suggestion:
```
        names = self.get_param_names()
        if self.action_command is not None:
            names[-1] = "action_command"
        return names
```